packages/esm-active-visits-app/src/visits-summary/visits-components/encounter-list.component.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState } from 'react';
+import DataTable, {
+  TableContainer,
+  Table,
+  TableHead,
+  TableExpandHeader,
+  TableRow,
+  TableHeader,
+  TableBody,
+  TableExpandRow,
+  TableCell,
+  TableExpandedRow,
+} from 'carbon-components-react/es/components/DataTable';
+import EncounterObservations from './encounter-observations.component';
+import styles from '../visit-detail-overview.scss';
+import { Observation } from '../visit.resource';
+import { useTranslation } from 'react-i18next';
+import { useLayoutType } from '@openmrs/esm-framework';
+
+interface EncounterListProps {
+  encounters: Array<{
+    id: any;
+    time: any;
+    encounterType: string;
+    provider: string;
+    obs: Array<Observation>;
+  }>;
+  visitUuid: string;
+}
+
+const EncounterListDataTable: React.FC<EncounterListProps> = ({ encounters, visitUuid }) => {
+  const { t } = useTranslation();
+  const layout = useLayoutType();
+  const isDesktop = layout === 'desktop';
+  const [headerWidth, setHeaderWidth] = useState(0);
+
+  const headerData = [
+    {
+      id: 1,
+      header: 'Time',
+      key: 'time',
+    },
+    {
+      id: 2,
+      header: 'Encounter Type',
+      key: 'encounterType',
+    },
+    {
+      id: 3,
+      header: 'Provider',
+      key: 'provider',
+    },
+  ];
+
+  useEffect(() => {
+    setHeaderWidth(document.getElementById(`header_${visitUuid}_0`)?.clientWidth);
```

Comment:
These ids are magic. Mind working with a ref and potentially a non-magic selector?